### PR TITLE
Fix completion for `@` and `@@` without ident

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Fixes <https://github.com/sorbet/sorbet/issues/5434>

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
cc @froydnj

File: parser/parser/cc/lexer.rl [hunk 1]
```diff
@@ -1334,6 +1334,7 @@ void lexer::set_state_expr_value() {
   # Interpolations with immediate variable names simply call into
   # the corresponding machine.
 
+  # TODO(jez) What is this for?
```
Comment (froydnj) [inline parser/parser/cc/lexer.rl:1337]: I assumed this was for string interpolation, though why we would need a completely separate rule for it is not obvious to me.

**Review by froydnj — COMMENTED** [review]
:tada:

File: parser/parser/cc/lexer.rl (hunk 1 above)
Comment (jez) [reply parser/parser/cc/lexer.rl]: luckily we had a test for this, so i fixed it.

**Review by froydnj — APPROVED** [review]

**Pull request merged**